plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -8375,4 +8493,19 @@ private Session withTimestampPrecision(Session session, HiveTimestampPrecision p
                 .setCatalogSessionProperty(catalog, "timestamp_precision", precision.name())
                 .build();
     }
+
+    private record BucketedFilterTestSetup(
+            String trinoTypeName,
```

Comment:
Instead of String for typename, can we use `Type` and then call getTypeId on that for generating the query ?

Reply:
It'll make things a bit complex. For example, in the case of map type, you'll have to do `new MapType(DOUBLE, BIGINT, new TypeOperators())`. Additionally, `getTypeId` for complex type like array will return `array(int)` instead of `array<int>`, so it won't work.

Reply:
I see, and other methods like getBaseName or getDisplayName don't fit here either ?

Reply:
yeah, they all return in a similar way

Reply:
Ok, let's keep it the way it is currently then